### PR TITLE
ci: Add Automated Linting file for Github Action

### PR DESCRIPTION
**Pull request by thenods**
Add Automated Linting file for Github Action

## Description
1. Move templates into `.github/templates`
2. Install `@typescript-eslint/parser`, `eslint `, `prettier `
3. Make a script for linting automatically

## How Has This Been Tested?
- CI script: tested with [act](https://github.com/nektos/act)
  - image

## Types of Changes

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+      
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      
+      - name: Get yarn cache directory path
```
Comment (WDever) [inline .github/workflows/lint.yml]: Cache yarn?

**Review by WDever — COMMENTED** [review]
Check the comment

File: .github/workflows/lint.yml (hunk 1 above)
Comment (thenods) [reply .github/workflows/lint.yml]: Deleted

**Review by WDever — APPROVED** [review]
LGTM

**Pull request merged**